### PR TITLE
Playwright e2e

### PR DESCRIPTION
**Pull request by Ectsang**
- [x] configured [GuardianUI](https://github.com/GuardianUI/GuardianTest) on top of [Playwright](https://playwright.dev/)
- [x] wrote initial test for impact and auth as examples

### CODE REVIEW

**Review by Syncretik — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -48,3 +48,10 @@ We use Tailwind for styling. As a rule of thumb we use use **component scoped st
 - add a reusable Tailwind component directive in `globals.css`
 
 _Avoid modifying the @base layer in `globals.css` (such as `span`, `p`, etc.) unless absolutely necessary. If doing so, make sure to smoke test element in app thoroughly to catch breaking changes._
+
+## End-to-end testing
+
+We use [GuardianUI](https://github.com/GuardianUI/GuardianTest) on top of [Playwright](https://playwright.dev/) for end-to-end testing. To set up for the first
+time, run `npx playwright install` after `npm install`. Then, to run the tests, run `npm run test:e2e` (or `npm run test:e2e:headless` for headless mode). Setting the `E2E_ENV` environment variable to production or test will run the tests against the production or test environment, respectively. By default, the tests will run against the local development environment. If you're developing on VSCode, you can install the [Playwright extension](https://playwright.dev/docs/getting-started-vscode) for ease of use.
```
Comment (Syncretik) [inline README.md:55]: change `npm run test:gui` to `npm run test:e2e` in `package.json`?

**Review by Syncretik — CHANGES_REQUESTED** [review]

File: tests/auth/auth.gui.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        "engineering.e2e@glodollar.org"
+      );
+      await page.getByTestId("submit-email-button").click();
+    }
+
+    expect(await page.isVisible("text=1,234.00")).toBeTruthy();
+  });
+
+  test("should allow social login", async ({ page, gui }) => {
+    const walletConnectButton = await page.isVisible("text=WalletConnect");
```
Comment (Syncretik) [inline tests/auth/auth.gui.ts:43]: think this one is duped incorrectly?

File: README.md (hunk 1 above)
Comment (Ectsang) [reply README.md:55]: Yep I'll change to `npm run test:e2e`. The tests failed as the text "Thanks for being part of the Glo movement" on the login modal was removed.

File: tests/auth/auth.gui.ts (hunk 2 above)
Comment (Ectsang) [reply tests/auth/auth.gui.ts:43]: Yes, changing the var name

File: lib/utils.ts [hunk 3]
```diff
@@ -40,6 +40,8 @@ export const api = () => apiInstance;
 
 export const isProd = () => process.env.NEXT_PUBLIC_VERCEL_ENV === "production";
 
+export const isE2E = () => process.env.E2E === "true";
```
Comment (Riglanto) [inline lib/utils.ts:43]: `NEXT_PUBLIC_` isn't needed here?

File: pages/_app.tsx [hunk 4]
```diff
@@ -13,12 +20,18 @@ import Analytics from "@/components/Analytics";
...
 import { GloSequenceConnector } from "@/lib/sequence-connector";
-import { getAllowedChains } from "@/lib/utils";
+
+import { isProd, isE2E } from "../lib/utils";
 
 import type { AppProps } from "next/app";
 
 const { chains, publicClient, webSocketPublicClient } = configureChains(
-  getAllowedChains(),
+  // Add conditional flag if IS_E2E_TEST? So we have polygon avialable as front one.
```
Comment (Riglanto) [inline pages/_app.tsx:29]: remove comment

side note: I'd suggest starting with isProd as it's most crucial path hence it's better to have it in beginning

File: tests/auth/auth.gui.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    const POLYGON = 137;
+    await gui.initializeChain(POLYGON, 45735806);
+    await page.goto("/");
+
+    await gui.setBalance(common.gloAddress, "12340000000000000000000");
+
+    await page.getByTestId("tos-checkbox").check();
+  });
+
+  // test("should allow email sign up", async ({ page, gui }) => {
```
Comment (Riglanto) [inline tests/auth/auth.gui.ts:21]: alsways good idea is too test.disable instead commenting

File: tests/index.gui.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
+// import { test } from "@guardianui/test";
```
Comment (Riglanto) [inline tests/index.gui.ts:1]: delete?

**Review by Riglanto — APPROVED** [review]

File: tests/auth/auth.gui.ts (hunk 5 above)
Comment (Ectsang) [reply tests/auth/auth.gui.ts:21]: using `test.skip`

File: lib/utils.ts (hunk 3 above)
Comment (Ectsang) [reply lib/utils.ts:43]: As discussed saving for future PR if needed

**Pull request merged**